### ISSUE TITLE
Dropbox backup issue

### ISSUE THREAD

**Issue by Collin7** [gcgarner/IOTstack#45]
Hi,

Just want to say thank you for the amazing work you have done on this project.

I'm not sure this is an issue so much as hoping you experienced the same during setup. The problem I am having is my dropbox back up. Running the scrips manually works well.  Using the cron tab and outputting to a log file, shows that it is working and uploading, but it never arrives on dropbox. Yet the log says uploading DONE.
This is only using crontab -e that does not work?
Any ideas?

**Comment by gcgarner** [comment]
Hello Collin

Just to confirm if you search for "docker" in the search box it doesn't return any results?

I have modified that backup script in the last few days (i removed the sudo in front of dropbox uploader and added some error handling if you dont have influxdb), please check that yours is up to date [script](https://github.com/gcgarner/IOTstack/blob/master/scripts/docker_backup.sh)

Have you uncommented the last line where it calls the dropbox-uploader?

**Comment by Collin7** [comment]
Hi Graham, I have indeed updated when I saw you updated the script.  I have also uncommented the line in dropbox uploader.  The strange thing is that the cron appears to work, instead of outputting to /dev/null. I output to a log file from the crontab to see what is happening, it prints out all the same stuff as when I ran it manually, till it completes uploading to dropbox, but if I check the date modified on dropbox, the file is not changed/uploaded.  I am thinking it might have something to do with permissions. I have made the scripts executable (not sure if required.)
Since it completes successfully, I am stumped.
If I run manually again it completes and the date modified in dropbox changes.

Iv tried chown for my user on all IOTstack directorys, iv tried chmod 777 (bad practice I know) just tried.

**Comment by gcgarner** [comment]
That is very odd indeed. I just scheduled mine to go off now and it was successful.
perhaps it is a permissions issue. 
what you could do is move of the docker.tar.gz into your home directory (this was a suggestion from @Paulf007 ) . chown the file with `chown pi:pi ~/docker.tar.gz` and then let the dropbox-uploader upload it.

in your crontab do you call the script with sudo or not? (preferably without sodu)

**Comment by Collin7** [comment]
So Im not sure what fixed it, but I deleted the script, pulled it again, and also deleted dropbox uploader and reinstalled it.  And not the back up is working from crontab.  Thanks for the help.

**Comment by Paulf007** [comment]
Just a thought but as you copy as the file as pi it should not be necessary to change the file permissions.
I have played with the scrip a few ways and in some cases, I have found that it will copy the files in the Apps folder of Dropbox with the folder extensions that you added so maybe have a look there?

Personally I have found that to make a daily backup is a bit of overkill unless you write a lot of data to the DB. So I do a backup once a week with a date in the file. Keep in mind that I do copy the files to a temp folder then upload it to dropbox and delete thereafter. I don't think that will kill the sd card. 
Writing data to a db for graphs is much more brutal.
I know the guys at EmonCms are doing the writing of the data through redis to reduce the read-writes but that's way above my head so I have never bothered with that. 
If you want to log data why not get a small online server, I use Virmac and at $2.25 for 512 mb ram it is not bad at all.
